backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -16,43 +16,54 @@
 
 import com.google.sps.data.Match;
 import com.google.sps.data.Participant;
+import com.google.sps.datastore.ParticipantDatastore;
 import java.time.Clock;
 import java.time.ZonedDateTime;
 import java.util.List;
+import javax.annotation.Nullable;
 
-/** Class used to find a match in a list of Participants with the most recently added Participant */
+/** Class used to find a match for new Participant with unmatched Participants in datastore */
 public final class FindMatchQuery {
 
   /** Maximum difference in duration to be compatible */
-  private static int MAX_DURATION_DIFF = 15;
-  /** extra padding time to ensure large enough meeting time block */
-  private static int PADDING_TIME = 15;
+  private static final int MAX_DURATION_DIFF = 15;
+  /** Extra padding time to ensure large enough meeting time block */
+  private static final int PADDING_TIME = 15;
   /** Reference clock */
-  private Clock clock;
+  private final Clock clock;
+  /** Datastore of Participants */
+  private final ParticipantDatastore participantDatastore;
 
   /** Constructor */
-  public FindMatchQuery(Clock clock) {
+  public FindMatchQuery(Clock clock, ParticipantDatastore participantDatastore) {
     this.clock = clock;
+    this.participantDatastore = participantDatastore;
   }
 
   /**
-   * Find match of new participant or add to list of participants, return whether or not match was
-   * found right after being added
+   * @return Match of new participant with unmatched participants by comparing duration and
+   *     availibility, or null if no match yet
    */
-  public Match findMatch(List<Participant> participants, Participant newParticipant) {
+  @Nullable
+  public Match findMatch(Participant newParticipant) {
```

Comment:
I moved add participant to datastore functionality to the AddParticipantServlet, but in this old commit, I actually added the participant to the datastore only after getting a list of unmatched participants from the datastore so that a participant would never be matched with themselves.